sign/rpmgensig.c

```diff
@@ -276,7 +296,17 @@ static int runGPG(sigTarget sigt, const char *sigfile)
     rpmPopMacro(NULL, "__plaintext_filename");
     rpmPopMacro(NULL, "__signature_filename");
 
-    fnamedPipe = Fopen(namedPipeName, "w");
+    while (1) {
+        errno = 0;
+        fnamedPipe = Fopen(namedPipeName, "w");
+        if (errno != EINTR)
```

Comment:
Never trust errno alone, it can get set as a side-effect even on success (generally speaking). You'll want to move the "Fopen failed" check from below here, and only look at errno if the open failed.